test/ntp_client_test.rb

```diff
@@ -0,0 +1,618 @@
+require_relative "test_helper"
+
+Yast.import "NtpClient"
+Yast.import "NetworkInterfaces"
+Yast.import "PackageSystem"
+Yast.import "Service"
+
+describe Yast::NtpClient do
+
+  subject { Yast::NtpClient }
+
+  describe "#Read" do
+    let(:data_dir) { File.join(File.dirname(__FILE__), "data") }
+
+    around do |example|
+      change_scr_root(File.join(data_dir, "scr_root_read"), &example)
+    end
+
+    before do
+      allow(subject).to receive(:Abort).and_return(false)
+      allow(Yast::NetworkInterfaces).to receive(:Read)
+      allow(Yast::Mode).to receive(:normal).and_return(false)
+      allow(Yast::Progress).to receive(:set)
+      allow(Yast::PackageSystem).to receive(:CheckAndInstallPackagesInteractive)
+        .with(["ntp"]).and_return(true)
+      allow(subject).to receive(:read_ad_address!)
+      allow(subject).to receive(:ProcessNtpConf)
+      allow(subject).to receive(:ReadSynchronization)
+      allow(subject).to receive(:read_chroot_config!)
+      allow(Yast::SuSEFirewall).to receive(:Read)
+      allow(Yast::Service).to receive(:Enabled).with("ntpd").and_return(true)
+    end
+
+    context "when config has been read previously" do
+      it "returns true" do
+        subject.config_has_been_read = true
+        expect(Yast::Mode).not_to receive(:normal)
+
+        expect(subject.Read).to eql(true)
+      end
+    end
+
+    context "when config has not been read" do
+      before do
+        subject.config_has_been_read = false
+      end
+
+      it "returns false if abort is pressed" do
+        allow(subject).to receive(:Abort).and_return(true)
+
+        expect(subject.Read).to eql(false)
+      end
+
+      it "doesn't show progress if it is not in normal Mode" do
+        expect(Yast::Progress).not_to receive(:New)
+
+        subject.Read
+      end
+
+      it "reads network interfaces config" do
+        expect(Yast::NetworkInterfaces).to receive(:Read)
+
+        subject.Read
+      end
+
+      context "when sysconfig network policity setting's config exists" do
+        it "sets ntp policy setting as read value" do
+          allow(Yast::FileChanges).to receive(:CheckFiles).with(["/etc/ntp.conf"])
+          allow(Yast::SCR).to receive(:Read)
+            .with(Yast::Path.new(".sysconfig.network.config.NETCONFIG_NTP_POLICY"))
+            .and_return("manual")
```

Comment:
I would use something like `some-policy`. Maybe it's set to `manual` in another piece of code and this test will keep passing.